File: 0392-is-subsequence/solution.swift
class Solution {
    func isSubsequence(_ s: String, _ t: String) -> Bool {
        var i=0
        var j=0

        let sChars = Array(s)
        let tChars = Array(t)

        while(i < s.count && j < t.count) {
            if(sChars[i] == tChars[j]) {
                i += 1
            }
            j += 1
        }
        return i == s.count
    }
}

File: 0392-is-subsequence/solution.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int n = s.length();
        int m = t.length();
        int i=0;
        int j=0;
        if(s.isEmpty()) return true;

        if(n > m) return false;
        while(i < n && j < m) {
            char ch1 = s.charAt(i);
            char ch2 = t.charAt(j);

            if(ch1 == ch2) {
                i++;
            }
                j++;
        }
        return i >= n;
    }
}
